docfx_yaml/extension.py

```diff
@@ -251,13 +252,18 @@ def _resolve_reference_in_module_summary(pattern, lines):
                 # Find the last component of the target. "~Queue.get" only returns <xref:get>
                 ref_name = ref_name[index:]
 
-            else:
+            elif pattern == REF_PATTERN_LAST:
                 index = matched_str.rfind('.') + 1
                 if index == 0:
                     # If there is no dot, push index to not include tilde
                     index = 1
                 ref_name = matched_str[index:]
 
+            else: #pattern == REF_PATTERN_BRACKETS:
```

Comment:
Make this `elif` and `raise ValueError` in the `else` block to indicate a programming mistake.